pyrk/th_system.py

```diff
@@ -104,3 +107,262 @@ def custom(self, t_b, t_env, res):
+                                            R=d["R"])
+                to_ret -= QconvBC/cap
+            if component.heatgen:
+                to_ret += self.heatgenFVM(component, power, omegas)/cap
+            for interface, d in component.cond.iteritems():
+                env = self.comp_from_name(interface)
+                Qcond = self.conductionFVM(component, env, t_idx)
+                to_ret -= Qcond/cap
+            for interface, d in component.conv.iteritems():
+                env = self.comp_from_name(interface)
+                if isinstance(env, THSuperComponent):
+                    Tr = env.compute_tr(component.T[t_idx].magnitude,
+                                        env.sub_comp[-2].T[t_idx].magnitude)
+                    Qconv = self.convection(t_b=component.T[t_idx].magnitude,
+                                            t_env=Tr,
+                                            h=d['h'],
+                                            A=d['area'])
+                    assert (Qconv*(component.T[t_idx].magnitude-Tr)) >= 0, '''
+                    convection from %s to %s, from low temperature %f to
+                    high temperature %f is wrong: %f''' % (
+                        component.name, env.name, component.T[t_idx].magnitude,
+                        Tr, Qconv.magnitude)
+                else:
+                    Qconv = self.convection(t_b=component.T[t_idx].magnitude,
+                                            t_env=env.T[t_idx].magnitude,
+                                            h=d['h'],
+                                            A=d['area'])
+                    assert (Qconv*(component.T[t_idx]-env.T[t_idx])).magnitude >= 0, \
+                        '''convection from %s to %s, from low temperature %f to
+                    high temperature %f is wrong: %f''' % (
+                        component.name, env.name, component.T[t_idx].magnitude,
+                        env.T[t_idx].magnitude, Qconv.magnitude)
+                to_ret -= Qconv/cap/component.vol.magnitude
+            for name, d in component.adv.iteritems():
+                Qadv = self.advection(t_out=component.T[t_idx].magnitude*2.0 -
+                                      d['t_in'].magnitude,
+                                      t_in=d['t_in'].magnitude,
+                                      m_flow=d['m_flow'],
+                                      cp=d['cp'])
+                to_ret -= Qadv/cap/component.vol.magnitude
+                assert Qadv >= 0, '''at step %d, %s is heating the system
+                by %f watts??? Tin is %f, tout is %f, tcomp is %f''' % (
+                    t_idx, component.name, Qadv, d['t_in'].magnitude,
+                    (component.T[t_idx]*2 - d['t_in']).magnitude,
+                    component.T[t_idx].magnitude)
+            return to_ret*units.kelvin/units.seconds
+
+    def BC_center(self, component, t_idx):
+        '''Qconduction from the center of a sphere to the first boundary
```

Comment:
Rather than variable names, like Qconduction, the docstring is a place to use words (like conductive heat production, in this case). I would recommend using a sentence here that answers the question "What does the BC_center function do and what value will it return". With what's here now, I can't quite tell what it does without reading the code.